expenses-manager-frontend/src/main/frontend/src/app/daily-stats/daily-stats.component.ts

```diff
@@ -75,18 +132,36 @@ export class DailyStatsComponent implements OnInit {
     this.endDate.setSeconds(59);
   }
 
-  previousDay(): void {
-    this.currentDate.setDate(this.currentDate.getDate() - 1);
-    this.currentDate = new Date(this.currentDate); 
-    this.fetchDataForSelectedDate(this.currentDate);
+  previousPeriod(): void {
+    if (this._selectedTab == "Year") {
+      this.currentDate.setFullYear(this.currentDate.getFullYear() - 1);
+      this.currentDate = new Date(this.currentDate);
+    }
+    if (this._selectedTab == "Month") {
+      this.currentDate.setMonth(this.currentDate.getMonth() - 1);
+      this.currentDate = new Date(this.currentDate);
+    }
+    if (this._selectedTab == "Day") {
+      this.currentDate.setDate(this.currentDate.getDate() - 1);
+      this.currentDate = new Date(this.currentDate);
+    }
```

Comment:
Cred ca mergea sa pui un switch aici, mai ales ca schimbi aceeasi variabila. Linia `this.currentDate = new Date(this.currentDate);` merge la final.